### PR TITLE
feat: request password reset

### PR DESCRIPTION
**Pull request by MikeSoft007**
​

## Description

This PR implements the password reset functionality, including the creation of a password reset link endpoint and an endpoint for users to enter a new password. The changes ensure users can reset their passwords securely and conveniently.

## Related Issue (Link to issue ticket)

This project only accepts pull requests related to open issues.
If suggesting a new feature or change, please discuss it in an issue first.
If fixing a bug, there should be an issue describing it with steps to reproduce.
Please link to the issue here: https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/496

## Motivation and Context
This change is required to enhance the user experience by allowing users to reset their passwords if forgotten. This feature solves the problem of users being unable to access their accounts due to forgotten passwords, reducing user frustration and improving engagement.

​

## How Has This Been Tested?
The functionality has been tested using pytest with the following test cases:

- [x] test_reset_password_success: Verifies successful password reset.
- [x] test_reset_password_invalid_token: Handles cases with an invalid reset token.
- [x] test_reset_password_user_not_found: Handles cases where the user is not found.
- [x] test_reset_password_passwords_do_not_match: Ensures passwords match during reset.
- [x] test_reset_password_database_error: Handles potential database errors.
- [x] test_create_valid_reset_link: Verifies the creation of a valid reset link.
- [x] test_create_reset_link_invalid_email: Handles cases with an invalid email for reset link creation.
All tests passed, confirming the functionality works as intended.

​

## Screenshots (if appropriate - Postman, etc):
image

​

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
      ​

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: api/v1/services/request_pwd.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        #     return success_response(
+        #         message="Password reset link sent successfully",
+        #         data=reset_link,
+        #         status_code=status.HTTP_201_CREATED
+        #     )
+        # return send_email
+
+        return success_response(
+                message="Password reset link sent successfully",
+                data=reset_link,
```
Comment (joboy-dev) [inline api/v1/services/request_pwd.py:74]: Change this to
```
data = {
    "reset_link": reset_link
}
```

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Make changes

**Comment by MikeSoft007** [comment]
@joboy-dev change updated

**Review by joboy-dev — APPROVED** [review]

**Review by Takinnuoye5 — APPROVED** [review]
Approved

**Review by Takinnuoye5 — APPROVED** [review]

**Review by Takinnuoye5 — APPROVED** [review]

**Pull request merged**